internal/metrics/remote_write_test.go

```diff
@@ -0,0 +1,253 @@
+package metrics_test
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/golang/snappy"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/project-flotta/flotta-device-worker/internal/metrics"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+var _ = Describe("remote write", func() {
+	var (
+		mockCtrl     *gomock.Controller
+		tsdbInstance *metrics.MockAPI
+		remoteWrite  *metrics.RemoteWrite
+		writeClient  *metrics.MockWriteClient
+		deviceID     = "deviceid"
+		minTime      time.Time
+	)
+
+	BeforeEach(func() {
+		minTime = time.Now()
+		os.Remove(metrics.LastWriteFileName)
+		mockCtrl = gomock.NewController(GinkgoT())
+		tsdbInstance = metrics.NewMockAPI(mockCtrl)
+		writeClient = metrics.NewMockWriteClient(mockCtrl)
+		remoteWrite = metrics.NewRemoteWrite("", deviceID, tsdbInstance)
+		remoteWrite.Client = writeClient
+	})
+
+	AfterEach(func() {
+		defer GinkgoRecover()
+		mockCtrl.Finish()
+		os.Remove(metrics.LastWriteFileName)
+	})
+
+	It("empty URL feature disabled", func() {
+		// given
+		remoteWrite.Client = nil
+
+		// when
+		remoteWrite.Write()
```

Comment:
I'm still not sure about this test. 
This test will fail only in case of `panic`, it doesn't guarantee that there are no invocation to TSBD and to the client.
I see the value of having a test that covers this scenario, I just don't think this is the right way.

@eloycoto WDYT?

Reply:
I would add a comment that no gomock expected call and should be ok

Reply:
1. testing that no panic happens is very important.
2. it does test that no invocation of TSDB and client happens. It does that by not expecting any calls.